File: frontend/src/types/attachment.ts
export type AttachmentOwnerType = "purchase" | "sale" | "transformation";

export interface Attachment {
  id: string;
  owner_type: AttachmentOwnerType;
  owner_id: string;
  original_filename: string;
  content_type: string;
  size_bytes: number;
  description: string | null;
  uploaded_by: string | null;
  uploaded_by_name: string | null;
  created_at: string;
}

export interface AttachmentListResponse {
  items: Attachment[];
  total: number;
}

/** Espejo de ALLOWED_ATTACHMENT_EXTENSIONS del backend. */
export const ATTACHMENT_ACCEPT = ".jpg,.jpeg,.png,.gif,.webp,.heic,.heif,.pdf";
export const MAX_ATTACHMENTS_PER_OWNER = 10;
export const MAX_ATTACHMENT_SIZE = 5 * 1024 * 1024;

File: backend/app/schemas/attachment.py
"""Schemas de Attachment — adjuntos de compras, ventas y transformaciones."""
from datetime import datetime
from typing import Literal, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

AttachmentOwner = Literal["purchase", "sale", "transformation"]


class AttachmentResponse(BaseModel):
    """Un adjunto. El contenido se pide aparte, por `/attachments/{id}/download`."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    owner_type: AttachmentOwner
    owner_id: UUID
    original_filename: str
    content_type: str
    size_bytes: int
    description: Optional[str] = None
    uploaded_by: Optional[UUID] = None
    uploaded_by_name: Optional[str] = None
    created_at: datetime


class AttachmentListResponse(BaseModel):
    items: list[AttachmentResponse]
    total: int


class AttachmentUpdate(BaseModel):
    """Solo la etiqueta es editable: el archivo se reemplaza borrando y subiendo."""

    description: Optional[str] = Field(None, max_length=200)
